src/client/http_handler.rs

```diff
@@ -183,27 +225,47 @@ impl HTTPHandler {
         .await
     }
 
-    pub(crate) async fn http_post(
+    pub(crate) async fn http_put(
         &self,
         uri: &str,
         payload: Option<Vec<u8>>,
         headers: HashMap<&str, &str>,
         deadline: Instant,
         ctrl_c: Arc<AtomicBool>,
-    ) -> Result<(String, u16), ClientError> {
-        self.http_do(uri, HttpVerb::Post, payload, headers, deadline, ctrl_c)
+    ) -> Result<(Pin<Box<ResultsStream>>, u16), ClientError> {
+        self.http_do(uri, HttpVerb::Put, payload, headers, deadline, ctrl_c)
             .await
     }
 
-    pub(crate) async fn http_put(
+    pub(crate) async fn http_post(
         &self,
         uri: &str,
         payload: Option<Vec<u8>>,
         headers: HashMap<&str, &str>,
         deadline: Instant,
         ctrl_c: Arc<AtomicBool>,
-    ) -> Result<(String, u16), ClientError> {
-        self.http_do(uri, HttpVerb::Put, payload, headers, deadline, ctrl_c)
+    ) -> Result<(Pin<Box<ResultsStream>>, u16), ClientError> {
```

Comment:
It might be nice to type alias this Pin<Box<ResultStream>> type

Reply:
Done